db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/PredicateList.java

```diff
@@ -3639,4 +3640,37 @@ public boolean isUnsatisfiable() {
                 return true;
         return false;
     }
+    @Override
+    public boolean canSupportIndexExcludedNulls(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td) throws StandardException {
+        return exclusionSupport(tableNumber,cd,td,true);
+    }
+
+    @Override
+    public boolean canSupportIndexExcludedDefaults(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td) throws StandardException {
+        return exclusionSupport(tableNumber,cd,td,false);
+    }
+
+    private boolean exclusionSupport(int tableNumber, ConglomerateDescriptor cd, TableDescriptor td, boolean excludeNulls) throws StandardException {
+        int size = size();
+        if (size == 0)
+            return false;
+        int[] baseColumnPositions = cd.getIndexDescriptor().baseColumnPositions();
+        ExcludedIndexColumnVisitor excludedIndexColumnVisitor = new ExcludedIndexColumnVisitor(0,baseColumnPositions[0],excludeNulls,excludeNulls?null:
```

Comment:
With table number set to 0, we may not be able to use the null/default exclusive index in joins. 

For example:

create table t1 (a1 int, b1 int, c1 int);
create table t2 (a2 int, b2 int, c2 int);
insert into t1 values (1,1,1), (2,2,2);
insert into t2 values (null,1,1), (2,2,2);

create index t2_ix on t2(a2) exclude null keys;

explain select a1, a2 from t1, t2 --splice-properties index=t2_ix
where b1=b2 and a2=2;

ERROR 42Y69: No valid execution plan was found for this statement. This is usually because an infeasible join strategy was chosen, or because an index was chosen which prevents the chosen join strategy from being used.

This is because in the above case, t2's table number is 1. 

By switching the order of the two tables, the index can be picked up.

It would be considered this as an enhancement in the near future if it is not going to be addressed in the first cut.